Sources/VLCMediaList.m

```diff
@@ -163,17 +164,18 @@ - (void)insertMedia:(VLCMedia *)media atIndex: (NSUInteger)index
     libvlc_media_list_insert_media(p_mlist, [media libVLCMediaDescriptor], (int)index);
 }
 
-- (void)removeMediaAtIndex:(NSUInteger)index
+- (VLCMedia *)removeMediaAtIndex:(NSUInteger)index
 {
+    VLCMedia *target = [_mediaObjects objectAtIndex:index];
+    
     dispatch_sync(_serialMediaObjectsQueue, ^{
-        if (index >= [_mediaObjects count])
-            return;
-        //remove from cached Media
+        // Remove from cached Media
```

Comment:
hmm, why do you remove the sanity check here? This is not a fail-safe operation.